specfile/specfile.py

```diff
@@ -71,9 +84,27 @@ def __init__(
                 Such sources include sources referenced from shell expansions
                 in tag values and sources included using the _%include_ directive.
         """
+        if stream is not None:
+            self._stream = stream
+            if stream.name and sourcedir is None:
+                raise ValueError("'sourcedir' must be defined")
+            elif stream.name is None:
+                self._path = None
+            elif stream.name is not None:
+                self._path = Path(stream.name)
+        elif path is not None and stream is None:
+            self._path = Path(path)
+            self._stream = self._path.open(
+                "r+", encoding="utf8", errors="surrogateescape"
+            )
+        else:
+            raise ValueError("Either 'stream' or 'path' must be provided")
+
+        self._is_writable = hasattr(self._stream, "write") and callable(
+            getattr(self._stream, "write", None)
+        )
```

Comment:
I would drop this attribute, as well as `_path`, both can be inferred from `_stream`. You can make it a property if you think it would be useful. The `path` property should definitely stay for backwards compatibility, but its setter must be adjusted - if a user changes path, `_stream` has to be updated/replaced.